Clean up ProjectFileInvalidator.findInvalidated a bit

jamesderlin opened the pull request:
In preparation for some refactoring that I will be doing to
`ProjectFileInvalidator.findInvalidated`, make its code a bit clearer:
* Indicate which arguments may be null.
* Don't bother calling `File.statSync` on the `.packages` file for the
  initial load.  This makes the checks for the `.packages` file
  consistent with those for other files.
* Use `DateTime.isAfter()` instead of comparing microseconds ourselves.

While I was touching this file, I also removed some unnecessary
comparisons to `false`. (`_ManifestAssetBundle.wasBuiltOnce()` is the
only implementation I can find of `AssetBundle.wasBuiltOnce()`, and it
never returns `null`.)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

jamesderlin committed:
Tighten up ProjectFileInvalidator.findInvalidated

* Indicate which arguments may be null.
* Don't bother calling File.statSync on the .packages file for the
  initial load.
* Use DateTime.isAfter() instead of comparing microseconds ourselves.

jamesderlin committed:
Remove unnecessary comparisons to true or false

jonahwilliams commented on packages/flutter_tools/lib/src/run_hot.dart:
```diff
@@ -1041,10 +1041,18 @@ class ProjectFileInvalidator {
 
   static List<Uri> findInvalidated({
-    @required DateTime lastCompiled,
+    @required DateTime /*?*/ lastCompiled,
```
nullable or not, I'd prefer we keep the comment based syntax out of the tool and wait for the real nnbd to be ready.

```suggestion
    @required DateTime lastCompiled,
```

jonahwilliams reviewed (approved):
LGTM wth nit

jamesderlin committed:
Update packages/flutter_tools/lib/src/run_hot.dart

Remove nullability hint.

Co-Authored-By: Jonah Williams <jonahwilliams@google.com>

jamesderlin committed:
Add a test for ProjectFileInvalidator.findInvalidated(lastCompiled: null, ...)

Add a test for ProjectFileInvalidator.findInvalidated with no last
compile time to try to appease the coverage test.
